### PR TITLE
Repository pt 2.

### PR DESCRIPTION
**Pull request by IvanKulundzic**
Implemented QuizRepository, layer between DataSource and UseCase.

### CODE REVIEW

File: QuizApp/App/Repository/QuizRepository.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    init(networkDataSource: QuizNetworkDataSourceProtocol, databaseDataSouce: QuizDatabaseDataSourceProtocol) {
+        self.networkDataSource = networkDataSource
+        self.databaseDataSouce = databaseDataSouce
+    }
+
+    @MainActor
+    func fetchQuizzes() async -> [QuizRepositoryModel] {
+        do {
+            let quizzes =  try await networkDataSource.fetchQuizes()
```
Comment (marinzg) [inline QuizApp/App/Repository/QuizRepository.swift]: Can you check on which thread are `networkDataSource` functions executed.

And you have one extra space.

File: QuizApp/App/Use cases/Quiz/Models/QuestionModel.swift [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import RealmSwift
```
Comment (marinzg) [inline QuizApp/App/Use cases/Quiz/Models/QuestionModel.swift]: Do you need `Realm` here?

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/App/Repository/QuizRepository.swift (hunk 1 above)
Comment (IvanKulundzic) [reply QuizApp/App/Repository/QuizRepository.swift]: This is the reason I used `@MainActor`.

`*** Terminating app due to uncaught exception 'RLMException', reason: 'Realm accessed from incorrect thread.'`

Instances of Realm and Object are thread-contained. They cannot be passed between threads or that exception will occur. So with `@MainActor,` instances are on the same thread. These are the threads used without `@MainActor`.

```
Thread on: <NSThread: 0x6000020a48c0>{number = 7, name = (null)}
Thread on: <NSThread: 0x6000020e11c0>{number = 4, name = (null)}
```

File: QuizApp/App/Repository/QuizRepository.swift [hunk 3]
```diff
@@ -22,17 +22,17 @@ final class QuizRepository: QuizRepositoryProtocol {
...
+                let quizzes = try await networkDataSource.fetchQuizes()
+                    .map { QuizRepositoryModel(from: $0) }
+                let quizzesToSave = try await networkDataSource.fetchQuizes()
+                    .map { QuizDatabaseModel(from: $0)}
+                saveQuizzes(quizzesToSave)
+                return quizzes
+            } catch {
+                return databaseDataSouce.fetchQuizzes()
+                    .map { QuizRepositoryModel(from: $0) }
+            }
```
Comment (marinzg) [inline QuizApp/App/Repository/QuizRepository.swift]: Remove one tab.

**Review by marinzg — CHANGES_REQUESTED** [review]

**Review by marinzg — APPROVED** [review]

**Pull request merged**